File: back/dentalium/src/main/java/com/romoreno/dentalium/infraestructure/adapter/in/controller/rest/medical_study/dto/DicomResponse.java
package com.romoreno.dentalium.infraestructure.adapter.in.controller.rest.medical_study.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@AllArgsConstructor
public class DicomResponse {

    private ListMedicalStudiesResponse.MedicalStudy medicalStudy;
    private List<DicomAttributes> dicomAttributes;
    private String base64Content;

    public DicomResponse() {
        dicomAttributes = new ArrayList<>();
    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class DicomAttributes {
        private String value;
        private String description;
    }


}

File: front/dentalium/src/app/model/api/DicomResponse.ts
export interface DicomResponse {
  medicalStudy: MedicalStudy;
  dicomAttributes: DicomAttribute[];
  base64Content: string;
}

export interface DicomAttribute {
  value: string;
  description: string;
}

export interface MedicalStudy {
  id: number;
  doctorId: number;
  studyType: StudyType;
  date: Date;
  filename: string;
}

export interface StudyType {
  id: number;
  description: string;
}
